### PR TITLE
Allow `loadStyleSheet` to accept an alternative injection point.

### PR DESCRIPTION
**Pull request by savetheclocktower**
This is my proposed solution to what we've been discussing in #21.

### CODE REVIEW

File: lib/browser/loader.js [hunk 1]
```diff
@@ -155,6 +157,15 @@ module.exports = {
 
     validateArguments(url, options, callback);
 
+    // This check doesn't happen in `validateArguments` because it's unique
+    // to `loadStyleSheet`.
+    if (
+      ('injectionNode' in options) &&
+      typeof options.injectionNode.nodeType !== 'number'
```
Comment (rmurphey) [inline lib/browser/loader.js]: Maybe better to just check that it has an `appendChild` method? idk.

Comment (savetheclocktower) [inline lib/browser/loader.js]: There are lots of ways to duck-type a DOM node and none of them stands out from the rest.

**Comment by rmurphey** [comment]
I'm good with this. There aren't docs on any of the options that `loadStyleSheet` or `loadScript` take -- maybe add:

```
### loadScript options

- **attributes**: an object containing key/value pairs to be used as
attributes on the created `<script>` element.
- **timeout**: the time in milliseconds before the loading should be
considered failed. Defaults to 1000ms.

### loadStyleSheet options

- **attributes**: an object containing key/value pairs to be used as
attributes on the created `<link>` element.
- **targetElement**: a DOM element into which the `<link>` element should be
placed. The default insertion point is after the first `<script>` element on the page.
- **timeout**: the time in milliseconds before the loading should be
considered failed. Defaults to 1000ms.
```

**Comment by savetheclocktower** [comment]
My last commit fleshed out the JSDoc, but I'll add that section to the Markdown documentation as well.

**Comment by rmurphey** [comment]
ping @lawnsea

File: lib/browser/loader.js (hunk 1 above)
Comment (lawnsea) [inline lib/browser/loader.js]: Do we need to duck-type it? Old IE doesn't support `HTMLElement`, but it does support `Element` and `Node`.

File: lib/browser/loader.js [hunk 2]
```diff
@@ -155,6 +163,15 @@ module.exports = {
 
     validateArguments(url, options, callback);
 
+    // This check doesn't happen in `validateArguments` because it's unique
+    // to `loadStyleSheet`.
+    if (
+      ('injectionNode' in options) &&
```
Comment (lawnsea) [inline lib/browser/loader.js]: Are these parens necessary? If not, I would prefer not to have them if we're not going to wrap the `typeof` check below.

**Comment by savetheclocktower** [comment]
All feedback addressed.

File: test/unit/lib/browser/loader.js [hunk 3]
```diff
@@ -303,6 +303,28 @@ module.exports = {
...
+            'must be a DOM node': function (test) {
+              test.throws(function () {
+                  loader.loadStyleSheet('/lib.loader.loadstylesheet.css', {
+                    injectionNode : false
+                  });
+                }, /`options.injectionNode` must be a DOM node/,
+                'should throw if `options.injectionNode` is not a DOM node'
+              );
+              test.done();
+            }
```
Comment (lawnsea) [inline test/unit/lib/browser/loader.js]: If possible, we should test the callback path as well. Perhaps we can stub the `appendChild` method on the `injectionNode` to `throw` when called?

**Comment by savetheclocktower** [comment]
Addressed Lon's feedback about test coverage.

**Comment by lawnsea** [comment]
:+1:

**Pull request closed without merging**